### ISSUE TITLE
[GR-60638][Native Image] JavaFX Maven - FileChooser generates error on runtime

### ISSUE THREAD

**Issue by valimaties** [oracle/graal#10073]
### Describe the Issue

This is my simple code, displaying a file chooser object in JavaFX Maven application:

```java
        setOnShowing(ev -> {
            Platform.runLater(() -> {
                FileChooser fileChooser = new FileChooser();
                fileChooser.getExtensionFilters().add(new FileChooser.ExtensionFilter("7-zip files (*.7z)", "*.7z"));
                fileChooser.setTitle("Choose a file name for backup file");
                var result = fileChooser.showSaveDialog(this);
                if (result == null) {
                    close();
                } else {
                    if (!result.getName().endsWith(".7z")) {
                        result.renameTo(new File(STR."\{result.getAbsolutePath()}.7z"));
                    }
                    backupController.startBackup(workingDirectory, result, mainWindow);
                }
            });
        });
```
I have added the necessary (I think) code in `reflect-config.json` file as:
```json
{
    "name": "com.sun.glass.ui.CommonDialogs",
    "allDeclaredFields": true,
    "allDeclaredMethods": true,
    "allDeclaredConstructors": true
  },
  {
    "name": "com.sun.glass.ui.win.WinCommonDialogs",
    "allDeclaredFields": true,
    "allDeclaredMethods": true,
    "allDeclaredConstructors": true
  }
```

I also added in `jni-config.json` file:
```json
{
    "name": "com.sun.glass.ui.win.WinCommonDialogs",
    "methods": [
      { "name": "_initIDs", "parameterTypes": [] },
      { "name": "showFileChooser", "parameterTypes": ["java.lang.String", "java.util.List", "int"] }
    ]
  },
  {
    "name": "com.sun.glass.ui.CommonDialogs",
    "methods": [
      { "name": "showFileChooser", "parameterTypes": ["com.sun.glass.ui.Window", "java.lang.String", "java.lang.String[]", "java.lang.String", "java.lang.String", "int"] },
      { "name": "createFileChooserResult", "parameterTypes": ["[Ljava/lang/String;", "[Lcom/sun/glass/ui/CommonDialogs$ExtensionFilter;", "int"] }
    ]
  }
```

### Using the latest version of GraalVM can resolve many issues.

- [ ] I tried with the [latest version](https://www.graalvm.org/downloads/) of GraalVM.

### GraalVM Version

openjdk version "22.0.2" 2024-07-16
OpenJDK Runtime Environment GraalVM CE 22.0.2+9.1 (build 22.0.2+9-jvmci-b01)
OpenJDK 64-Bit Server VM GraalVM CE 22.0.2+9.1 (build 22.0.2+9-jvmci-b01, mixed mode, sharing)

### Operating System and Version

Windows 11 x64

### Build Command

using profile

### Expected Behavior

to display the actual FileChooser Dialog box.

### Actual Behavior

not working, and I don't know what to do anymore.. I search entire internet of a solution, I did not found anything 

### Steps to Reproduce

I added the code below

### Additional Context

This is the maven profile I use in my pom file, and I package my app using maven's `package -f pom.xml -Pnative`

```java
<profiles>
        <profile>
            <id>native</id>
            <build>
                <plugins>
                    <plugin>
                        <groupId>org.graalvm.buildtools</groupId>
                        <artifactId>native-maven-plugin</artifactId>
                        <version>${native.maven.plugin.version}</version>
                        <executions>
                            <execution>
                                <id>build-native</id>
                                <goals>
                                    <goal>compile-no-fork</goal>
                                </goals>
                                <phase>package</phase>
                            </execution>
                        </executions>
                        <configuration>
                            <mainClass>com.myProject.test.MyLauncher</mainClass>
                            <fallback>false</fallback>
                            <buildArgs>
                                <buildArg>--no-fallback</buildArg>
                                <buildArg>--enable-preview</buildArg>
                                <buildArg>--report-unsupported-elements-at-runtime</buildArg>
                                <buildArg>-H:+ReportExceptionStackTraces</buildArg>
                                <buildArg>-H:+AddAllCharsets</buildArg>
                                <buildArg>--initialize-at-build-time=org.slf4j,ch.qos.logback</buildArg>
                                <buildArg>--initialize-at-run-time=javafx,javafx.scene.control.Control,javafx.scene.control.skin.TextFieldSkin,com.sun.glass.ui.CommonDialogs,com.sun.glass.ui.win.WinCommonDialogs</buildArg>
                                <buildArg>-H:+UnlockExperimentalVMOptions</buildArg>
                                <buildArg>-H:EnableURLProtocols=http,https,jar,jrt</buildArg>
                                <buildArg>-H:+PrintClassInitialization</buildArg>
                                <buildArg>-H:ReflectionConfigurationFiles=${project.basedir}/src/main/resources/META-INF/native-image/com/myProject/test/reflect-config.json</buildArg>
                                <buildArg>-H:ResourceConfigurationFiles=${project.basedir}/src/main/resources/META-INF/native-image/com/myProject/test/resource-config.json</buildArg>
                                <!--
                                <buildArg>-H:NativeLinkerOption=/SUBSYSTEM:WINDOWS</buildArg> &lt;!&ndash; This arg will stop running console windows at startup &ndash;&gt;
                                <buildArg>-H:NativeLinkerOption=/ENTRY:mainCRTStartup</buildArg> &lt;!&ndash; This arg will stop running console windows at startup &ndash;&gt;
                                -->
                                <buildArg>-H:NativeLinkerOption="${project.basedir}/src/main/resources/appIcon.res"</buildArg> <!-- add icon -->
                                <buildArg>-J-Xmx4g</buildArg>
                            </buildArgs>
                            <agent>
                                <enabled>true</enabled>
                                <options>
                                    <option>experimental-class-loader-support</option>
                                </options>
                            </agent>
                        </configuration>
                    </plugin>
                </plugins>
            </build>
        </profile>
    </profiles>
```

native-maven-plugin-version is a property in properties, and its value is 0.10.3

### Build Log Output and Error Messages

This is the error from console, as I commented the lines which hide the console at execution, in profile:

```java
SEVERE: Uncaught exception in thread: JavaFX Application Thread
java.lang.NoSuchMethodError: com.sun.glass.ui.CommonDialogs.createFileChooserResult([Ljava/lang/String;[Lcom/sun/glass/ui/CommonDialogs$ExtensionFilter;I)Lcom/sun/glass/ui/CommonDialogs$FileChooserResult;
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions$Support.getMethodID(JNIFunctions.java:1347)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions$Support.getMethodID(JNIFunctions.java:1332)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions.GetStaticMethodID(JNIFunctions.java:440)
        at com.sun.glass.ui.win.WinCommonDialogs._initIDs(Native Method)
        at com.sun.glass.ui.win.WinCommonDialogs.<clinit>(WinCommonDialogs.java:40)
        at com.sun.glass.ui.win.WinApplication.staticCommonDialogs_showFileChooser(WinApplication.java:305)
        at com.sun.glass.ui.CommonDialogs.showFileChooser(CommonDialogs.java:213)
        at com.sun.javafx.tk.quantum.QuantumToolkit.showFileChooser(QuantumToolkit.java:1727)
        at javafx.stage.FileChooser.showDialog(FileChooser.java:396)
        at javafx.stage.FileChooser.showSaveDialog(FileChooser.java:372)
        at ro.mvmusicsoft.yemcefx.Stages.Backup.lambda$new$2(Backup.java:68)
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:456)
        at java.base@22.0.2/java.security.AccessController.executePrivileged(AccessController.java:128)
        at java.base@22.0.2/java.security.AccessController.doPrivileged(AccessController.java:400)
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:455)
        at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
        at com.sun.glass.ui.win.WinApplication._enterNestedEventLoopImpl(Native Method)
        at com.sun.glass.ui.win.WinApplication._enterNestedEventLoop(WinApplication.java:211)
        at com.sun.glass.ui.Application.enterNestedEventLoop(Application.java:513)
        at com.sun.glass.ui.EventLoop.enter(EventLoop.java:107)
        at com.sun.javafx.tk.quantum.QuantumToolkit.enterNestedEventLoop(QuantumToolkit.java:656)
        at javafx.stage.Stage.showAndWait(Stage.java:469)
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:456)
        at java.base@22.0.2/java.security.AccessController.executePrivileged(AccessController.java:128)
        at java.base@22.0.2/java.security.AccessController.doPrivileged(AccessController.java:400)
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:455)
        at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
        at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
        at com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:184)
        at java.base@22.0.2/java.lang.Thread.runWith(Thread.java:1583)
        at java.base@22.0.2/java.lang.Thread.run(Thread.java:1570)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:853)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:829)
```

However, I searched in `FileChooser` class and `CommonDialogs` class and there is no `createFileChooserResults` method. Why this error?!

**Comment by valimaties** [comment]
I tryed using DirectoryChooser, but same thing, same error, even changing `GraalVM` to `23.0.1`:

```java
SEVERE: Uncaught exception in thread: JavaFX Application Thread
java.lang.NoSuchMethodError: com.sun.glass.ui.CommonDialogs.createFileChooserResult([Ljava/lang/String;[Lcom/sun/glass/ui/CommonDialogs$ExtensionFilter;I)Lcom/sun/glass/ui/CommonDialogs$FileChooserResult;
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions$Support.getMethodID(JNIFunctions.java:1851)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions$Support.getMethodID(JNIFunctions.java:1836)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions.GetStaticMethodID(JNIFunctions.java:449)
        at com.sun.glass.ui.win.WinCommonDialogs._initIDs(Native Method)
        at com.sun.glass.ui.win.WinCommonDialogs.<clinit>(WinCommonDialogs.java:40)
        at com.sun.glass.ui.win.WinApplication.staticCommonDialogs_showFolderChooser(WinApplication.java:312)
        at com.sun.glass.ui.CommonDialogs.showFolderChooser(CommonDialogs.java:227)
        at com.sun.javafx.tk.quantum.QuantumToolkit.showDirectoryChooser(QuantumToolkit.java:1758)
        at javafx.stage.DirectoryChooser.showDialog(DirectoryChooser.java:109)
        at ro.mvmusicsoft.yemcefx.Stages.Backup.lambda$new$2(Backup.java:85)
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:456)
        at java.base@23.0.1/java.security.AccessController.executePrivileged(AccessController.java:132)
        at java.base@23.0.1/java.security.AccessController.doPrivileged(AccessController.java:400)
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:455)
        at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
        at com.sun.glass.ui.win.WinApplication._enterNestedEventLoopImpl(Native Method)
        at com.sun.glass.ui.win.WinApplication._enterNestedEventLoop(WinApplication.java:211)
        at com.sun.glass.ui.Application.enterNestedEventLoop(Application.java:513)
        at com.sun.glass.ui.EventLoop.enter(EventLoop.java:107)
        at com.sun.javafx.tk.quantum.QuantumToolkit.enterNestedEventLoop(QuantumToolkit.java:656)
        at javafx.stage.Stage.showAndWait(Stage.java:469)
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:456)
        at java.base@23.0.1/java.security.AccessController.executePrivileged(AccessController.java:132)
        at java.base@23.0.1/java.security.AccessController.doPrivileged(AccessController.java:400)
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:455)
        at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
        at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
        at com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:184)
        at java.base@23.0.1/java.lang.Thread.runWith(Thread.java:1588)
        at java.base@23.0.1/java.lang.Thread.run(Thread.java:1575)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:832)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:808)
```

**Comment by oubidar-Abderrahim** [comment]
Please provide a complete reproducer for this issue; Thank you

**Comment by valimaties** [comment]
https://github.com/valimaties/reproducer

I created a project from scratch, based on default maven Hello World.
I added a FileChooser in HelloController, to open the file chooser dialog. 
However, even if I set the same settings as in my other project, for which I started this issue, for this one the exe is generated, however it has some errors on start.
It runs very well from IntelliJ but not the native-image created.

Note that in buildArgs on the profile there is a path to javafx-sdk-23.0.1. 

Please tell me what I'm doing wrong, as I set everything as the GraalVM documentation sais.

Used  Graalvm 23.0.1 SDK.

**Comment by valimaties** [comment]
any news? is there a way to use graalvm for maven with full integration in a easier way?

**Comment by valimaties** [comment]
Ok... seems no one give me an answer, I think nobody uses Maven and built a native-image using Graalvm. This is sad for me, as I wrote a lot of code in my project without knowing that is not a way to build a simple native-image in java. Ohh... 😕

**Comment by oubidar-Abderrahim** [comment]
If you need examples for using native build tools, you can check our GraalVM demos repo: https://github.com/graalvm/graalvm-demos/tree/master/native-image/native-build-tools
For the example you've shared, I tried building the image using `mvn clean package -Pnative` and encountered the same error, we'll investigate this further and get back to you

**Comment by valimaties** [comment]
Thank you, I hope for good news 😊

**Comment by oubidar-Abderrahim** [comment]
I believe this is fixed in the next release of GraalVM for JDK24 and javaFX 24-ea+19

**Comment by valimaties** [comment]
It still does not work. I downloaded graalvm 24 and also javafx 24 early access. 
I'm unable to build the native-image for such a small project! :( This is very sad, as long as I have a big maven project which I expected to be built as native-image, but as I see my chance to do that is becoming to almost zero. 

Why this process has to be so hard?! Why all those parameters in pom file for the plugin? Why not just a simple build instruction and the engine to build the native-image?! We must do a lot of hard work, personally I do this for three months already and I expected a solution for just this simple project, and I have not! 
I probably must think of get rest, I worked a lot for my big project and now I have to make some decision about leaving java forever and search for a language which can achieve the needed result!

Thanks for ... whatever! :( Ohhh

**Comment by valimaties** [comment]
I've tried last from the scratch, using graalvm 24 and javafx 24 EA. I tried setting up the profile in pom file as you shared on the [examples webpage](https://www.graalvm.org/latest/reference-manual/native-image/guides/use-reachability-metadata-repository-maven/) including reachability-metadata.json file, but as I see those examples are no more relevant, they simply don't work. For example exec does not exists anymore in goals, so the information on that webpage is not anymore current affairs. 

So, definitelly, with old information the result cannot be achieved!

**Comment by valimaties** [comment]
Nothing new? I tried also the 24 version with JavaFX 24 EA and the same thing...

**Comment by valimaties** [comment]
Issue started on nov 2024, and nothing new about this issue? Is very strange. I'm doing something wrong? I need something installed which I don't have? No one say nothing! Sad 😒

**Comment by droiddk** [comment]
@valimaties have you tried the Gluon version? https://github.com/gluonhq/graal/releases/tag/gluon-23%2B25.1-dev-2409082136
I use that version with JavaFX FileChoser without any errors.

Native image with JavaFX is very easy using GluonFX, just make sure to follow the docs 100%:

1. https://docs.gluonhq.com/#platforms_windows
2. https://docs.gluonhq.com/#_gluonfx_plugin_for_maven

**Comment by valimaties** [comment]
I remember I used GluonFX to build native-image from my project.
But my project is very consistent, it has a lot of classes and uses a bounch of external dependencies. It didn't worked with GluonFX, either. I found commented dependencies in my pom file, in the git repository.

My project also uses media, because the user can load and play some audio samples (aiff files), it uses animations, seven-zip amd a lot of other things.
FileChooser was only a single piece of my project which broke the building process. I bet there are a lot of others.

I'm not so advanced in Maven "undergrounds" and, for me, at least, it is a pain to do such a hard work to build native-image. I thought there is only a plugin which I will add it to my project and it will do everything for me, without putting such a big effort, to add a lot of parameters to it, and so on... Ohhh, very hard process!

**Comment by vjovanov** [comment]
This looks like there is a missing entry in the metadata. I would run with the latest version of GraalVM ([ideally early access for GraalVM 25](https://github.com/graalvm/oracle-graalvm-ea-builds/releases)). Combine that with the flags proposed in our [troubleshooting guide](https://www.graalvm.org/latest/reference-manual/native-image/guides/troubleshoot-run-time-errors/) and the missing metadata should be displayed.

**Comment by valimaties** [comment]
Hi @vjovanov , thanks for your comment. I will try your suggestion.

**Comment by valimaties** [comment]
> This looks like there is a missing entry in the metadata. I would run with the latest version of GraalVM ([ideally early access for GraalVM 25](https://github.com/graalvm/oracle-graalvm-ea-builds/releases)). Combine that with the flags proposed in our [troubleshooting guide](https://www.graalvm.org/latest/reference-manual/native-image/guides/troubleshoot-run-time-errors/) and the missing metadata should be displayed.

I've tried GraalVM 25 (EA), latest in this moment, graalvm-jdk-25+37.1. I get some other errors when I try to build the native-image. Trying GraalVM 26 is even worst, because it seems it missing jdk.jsobject from mods, this being one of the errors when I try to build. 

Nothing works, a lot of errors, when I solve one, I get another one. So, for me at least, I don't think GraalVM is doing something regarding native-image from maven projects! 
I have almost one year lost expecting something will be done with this issue, but I gues GraalVM is not even close to be a solution of building native-images from Maven projects.

So, what remains is to start from scratch in other programming language which is able to build a native-image.

**Comment by vjovanov** [comment]
These are likely missing metadata errors. Can you post the errors here?

Also, you can try the new `-H:Preserve=all` if you don't want to spend time with adding metadata. With `-H:Preserve=all` the image will be larger of course, but it should work.

**Comment by valimaties** [comment]
`-H:Preserve=all` seems to eat all my RAM memory (16GB on PC, so it eats all available RAM and process crushes).
I need to find a more rocket PC, maybe my workplace server, which have more RAM and a better processor.

**Comment by vjovanov** [comment]
You can also narrow it down with:
1. `-H:Preserve=module=<module>` preserves all elements from a given module
2. `-H:Preserve=module=ALL-UNNAMED` preserves all elements from all class-path entries
3. `-H:Preserve=package=<package>` preserves all elements from a given package

`all` includes the whole JDK.

**Comment by valimaties** [comment]
```
Build resources:
 - 10.77GB of memory (63.6% of system memory, using available memory)
 - 8 thread(s) (100.0% of 8 available processor(s), determined at start)
Terminating due to java.lang.OutOfMemoryError: Java heap space
The Native Image build process ran out of memory.
Please make sure your build system has more memory available.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  45:29 min
[INFO] Finished at: 2025-09-25T14:32:10+03:00
```

😄 and 😭in the same time, after 45 minutes of waiting, and closing all my other software, to free up memory, I got this error 😒
How much memory do I have to own on my PC to make it work to final step?

**Comment by vjovanov** [comment]
You need between 24 and 32 GB for most projects. For very large projects (400+ MB of JARs) you will need 48 GB.

Note that you can use the `-Ob` flag that will reduce the memory and time significantly.

**Comment by valimaties** [comment]
> You need between 24 and 32 GB for most projects. For very large projects (400+ MB of JARs) you will need 48 GB.

Is this documented somewhere? Thanks

**Comment by vjovanov** [comment]
One correction, for 24 to 32 GB is enough for most medium-sized projects. Small projects go well with just a few GB of ram.

I don't think we have this documented as the memory requirements change often. It would be interesting to warn the user if we detect to many types for the given memory size.

CC @ban-mi

**Comment by valimaties** [comment]
What is strange is that my project built as jar have aprox. 6 MB. For a 6MB project to use aprox 8GB of RAM, to me is very strange!

![image](https://github.com/user-attachments/assets/f74c0573-9986-4e0f-855c-628cf7604dc6)

Sorry for picture take with phone 😄

**Comment by vjovanov** [comment]
Maybe the JVM took more RAM than necessary. You can try with `-J-Xmx4g` and I believe it will work.

**Comment by valimaties** [comment]
> Maybe the JVM took more RAM than necessary. You can try with `-J-Xmx4g` and I believe it will work.

I've tried it and after almost 7 hours of waiting I need to stop it.

I've just bought a RAM memory kit, to double the amount of RAM on my PC, so I will try with 32 GB of RAM next days. If neither will work, then is something wrong with GraalVM, it is not possible to do so in such a small project. It is both hilarious and sad.

I'll just come back with feedback.

Regards.

**Comment by vjovanov** [comment]
Thanks @valimaties for going an extra mile! I would say that there must be some other issue. Maybe also some of the libraries are using up resources in their features.

**Comment by valimaties** [comment]
Hi!
Feedback: (😐)

After I mount two DIMMs of 8GB each, now I have 32GB of RAM.

In profile args I used 24GB of my RAM to be used by package. 
If I use `<buildArg>-H:Preserve=module=ALL-UNNAMED</buildArg>` it builds the exe file in aprox. 15 minutes, with a size of 680MB. But when I start it, I get some missing reachability `glob` things, which I must add them. And it keeps telling me of some log4j-test##$#$.**** classes to add into reachability.

Then I tried with `<buildArg>-H:Preserve=all</buildArg>` . In this step I'll get the following errors in build process:

```
Build resources:
 - 22.91GB of memory (67.1% of system memory, set via '-Xmx24g')
 - 8 thread(s) (100.0% of 8 available processor(s), determined at start)
[2/8] Performing analysis...  [*****]                                                                (1097.6s @ 18.06GB)
  170,252 types, 203,021 fields, and 578,641 methods found reachable
  387,949 types, 187,117 fields, and 415,074 methods registered for reflection
   51,302 types, 186,963 fields, and 412,452 methods registered for JNI access
        0 downcalls and 0 upcalls registered for foreign access
        5 native libraries: crypt32, ncrypt, psapi, version, winhttp

=== Image generator watchdog detected no activity. This can be a sign of a deadlock during image building. Dumping all stack traces. Current time: Tue Oct 07 10:11:26 EEST 2025
"main" Id=3 in WAITING on lock=java.util.concurrent.ForkJoinTask$AdaptedRunnableAction@4bd9b14e
    at java.base@25/jdk.internal.misc.Unsafe.park(Native Method)
    at java.base@25/java.util.concurrent.locks.LockSupport.park(LockSupport.java:369)
    at java.base@25/java.util.concurrent.ForkJoinTask.awaitDone(ForkJoinTask.java:445)
    at java.base@25/java.util.concurrent.ForkJoinTask.awaitDone(ForkJoinTask.java:500)
    at java.base@25/java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:666)
    at app/org.graalvm.nativeimage.builder/com.oracle.svm.hosted.meta.UniverseBuilder.build(UniverseBuilder.java:209)
    at app/org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:594)
    at app/org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:535)
    at app/org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:548)
    at app/org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:761)
    at app/org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.start(NativeImageGeneratorRunner.java:163)
    at app/org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:111)

"Reference Handler" Id=7 in RUNNABLE
    at java.base@25/java.lang.ref.Reference.waitForReferencePendingList(Native Method)
    at java.base@25/java.lang.ref.Reference.processPendingReferences(Reference.java:246)
    at java.base@25/java.lang.ref.Reference$ReferenceHandler.run(Reference.java:208)

"Finalizer" Id=8 in WAITING on lock=java.lang.ref.ReferenceQueue$Lock@edb2651
    at java.base@25/java.lang.Object.wait0(Native Method)
    at java.base@25/java.lang.Object.wait(Object.java:389)
    at java.base@25/java.lang.Object.wait(Object.java:351)
    at java.base@25/java.lang.ref.ReferenceQueue.remove0(ReferenceQueue.java:137)
    at java.base@25/java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:215)
    at java.base@25/java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:165)

"Signal Dispatcher" Id=9 in RUNNABLE

"Attach Listener" Id=10 in RUNNABLE

"Common-Cleaner" Id=17 in TIMED_WAITING on lock=java.lang.ref.ReferenceQueue$Lock@4164ed9e
    at java.base@25/java.lang.Object.wait0(Native Method)
    at java.base@25/java.lang.Object.wait(Object.java:389)
    at java.base@25/java.lang.ref.ReferenceQueue.remove0(ReferenceQueue.java:123)
    at java.base@25/java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:201)
    at java.base@25/jdk.internal.ref.CleanerImpl.run(CleanerImpl.java:146)
    at java.base@25/java.lang.Thread.runWith(Thread.java:1487)
    at java.base@25/java.lang.Thread.run(Thread.java:1474)
    at java.base@25/jdk.internal.misc.InnocuousThread.run(InnocuousThread.java:148)

"Notification Thread" Id=18 in RUNNABLE

"native-image pid watcher" Id=19 in TIMED_WAITING on lock=java.util.TaskQueue@3f1f0ca4
    at java.base@25/java.lang.Object.wait0(Native Method)
    at java.base@25/java.lang.Object.wait(Object.java:389)
    at java.base@25/java.util.TimerThread.mainLoop(Timer.java:569)
    at java.base@25/java.util.TimerThread.run(Timer.java:522)

"Thread-1" Id=20 in RUNNABLE
    at java.management@25/sun.management.ThreadImpl.dumpThreads0(Native Method)
    at java.management@25/sun.management.ThreadImpl.dumpAllThreads(ThreadImpl.java:505)
    at java.management@25/sun.management.ThreadImpl.dumpAllThreads(ThreadImpl.java:493)
    at app/org.graalvm.nativeimage.builder/com.oracle.svm.hosted.DeadlockWatchdog.threadDump(DeadlockWatchdog.java:135)
    at app/org.graalvm.nativeimage.builder/com.oracle.svm.hosted.DeadlockWatchdog.reportFailureState(DeadlockWatchdog.java:103)
    at app/org.graalvm.nativeimage.builder/com.oracle.svm.hosted.DeadlockWatchdog.watchdogThread(DeadlockWatchdog.java:86)
    at app/org.graalvm.nativeimage.builder/com.oracle.svm.hosted.DeadlockWatchdog$$Lambda/0x00000000171c5d98.run(Unknown Source)
    at java.base@25/java.lang.Thread.runWith(Thread.java:1487)
    at java.base@25/java.lang.Thread.run(Thread.java:1474)

"ForkJoinPool.commonPool-worker-1" Id=29 in RUNNABLE
    at java.base@25/java.util.ArrayList.add(ArrayList.java:487)
    at java.base@25/java.util.ArrayList.add(ArrayList.java:497)
    at app/org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.results.StrengthenGraphs$StrengthenSimplifier.strengthenStampFromTypeFlow(StrengthenGraphs.java:1013)
    at app/org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.results.StrengthenGraphs$StrengthenSimplifier.simplify(StrengthenGraphs.java:449)
    at platform/jdk.graal.compiler/jdk.graal.compiler.phases.common.CanonicalizerPhase.tryCanonicalize(CanonicalizerPhase.java:597)
    at platform/jdk.graal.compiler/jdk.graal.compiler.phases.common.CanonicalizerPhase.processNode(CanonicalizerPhase.java:504)
    at platform/jdk.graal.compiler/jdk.graal.compiler.phases.common.CanonicalizerPhase.processWorkSet(CanonicalizerPhase.java:349)
    at platform/jdk.graal.compiler/jdk.graal.compiler.phases.common.CanonicalizerPhase.run(CanonicalizerPhase.java:246)
    at platform/jdk.graal.compiler/jdk.graal.compiler.phases.common.CanonicalizerPhase.run(CanonicalizerPhase.java:92)
    at platform/jdk.graal.compiler/jdk.graal.compiler.phases.BasePhase.apply(BasePhase.java:468)
    at platform/jdk.graal.compiler/jdk.graal.compiler.phases.BasePhase.apply(BasePhase.java:334)
    at app/org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.results.StrengthenGraphs$AnalysisStrengthenGraphsPhase.run(StrengthenGraphs.java:336)
    at app/org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.results.StrengthenGraphs$AnalysisStrengthenGraphsPhase.run(StrengthenGraphs.java:322)
    at platform/jdk.graal.compiler/jdk.graal.compiler.phases.BasePhase.apply(BasePhase.java:468)
    at platform/jdk.graal.compiler/jdk.graal.compiler.phases.BasePhase.apply(BasePhase.java:334)
    at app/org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.results.StrengthenGraphs.applyResults(StrengthenGraphs.java:263)
    at app/org.graalvm.nativeimage.builder/com.oracle.svm.hosted.meta.UniverseBuilder.lambda$buildProfilingInformation$0(UniverseBuilder.java:431)
    at app/org.graalvm.nativeimage.builder/com.oracle.svm.hosted.meta.UniverseBuilder$$Lambda/0x000000001dc2baa0.accept(Unknown Source)
    at java.base@25/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:186)
    at java.base@25/java.util.HashMap$ValueSpliterator.forEachRemaining(HashMap.java:1788)
    at java.base@25/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:570)
    at java.base@25/java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:293)
    at java.base@25/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:759)
    at java.base@25/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:511)
    at java.base@25/java.util.concurrent.ForkJoinPool.helpComplete(ForkJoinPool.java:2375)
    at java.base@25/java.util.concurrent.ForkJoinTask.awaitDone(ForkJoinTask.java:497)
    at java.base@25/java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:666)
    at java.base@25/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:681)
    at java.base@25/java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:162)
    at java.base@25/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:176)
    at java.base@25/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:264)
    at java.base@25/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:632)
    at java.base@25/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:806)
    at app/org.graalvm.nativeimage.builder/com.oracle.svm.hosted.meta.UniverseBuilder.buildProfilingInformation(UniverseBuilder.java:427)
    at app/org.graalvm.nativeimage.builder/com.oracle.svm.hosted.meta.UniverseBuilder$$Lambda/0x000000001dc2b870.run(Unknown Source)
    at java.base@25/java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1593)
    at java.base@25/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:511)
    at java.base@25/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1450)
    at java.base@25/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2019)
    at java.base@25/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:187)

... and more like above
```

some instructions to me?

Thanks!

**Edit**

After some small changes, using `Preserve=all`, and changed the memory size from 24g to 26g, the results is an executable file of 1.65GB (LOL). After I start it I've still get some missing things in resources:

```
Exception in thread "main" com.oracle.svm.core.jdk.resources.MissingResourceRegistrationError: Cannot access resource at path 'log4j2-test6b143ee9.yaml'. To allow this operation, add the following to the 'resources' section of 'reachability-metadata.json' and rebuild the native image:

  {
    "glob": "log4j2-test6b143ee9.yaml"
  }
The 'reachability-metadata.json' file should be located in 'META-INF/native-image/<group-id>/<artifact-id>/' of your project. For further help, see https://www.graalvm.org/latest/reference-manual/native-image/metadata/#resources
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.resources.MissingResourceRegistrationUtils.reportResourceAccess(MissingResourceRegistrationUtils.java:59)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.Resources.createURLs(Resources.java:704)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.Resources.createURL(Resources.java:625)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.Resources.createURL(Resources.java:617)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.ResourcesHelper.nameToResourceURL(ResourcesHelper.java:42)
        at java.base@25/jdk.internal.loader.BootLoader.findResource(BootLoader.java:678)
        at java.base@25/java.lang.ClassLoader.getResource(ClassLoader.java:1346)
        at java.base@25/java.lang.ClassLoader.getResource(ClassLoader.java:1344)
        at java.base@25/java.lang.ClassLoader.getSystemResource(ClassLoader.java:1615)
        at org.apache.logging.log4j.core.util.Loader.getResource(Loader.java:130)
        at org.apache.logging.log4j.core.config.ConfigurationSource.fromResource(ConfigurationSource.java:351)
        at org.apache.logging.log4j.core.config.ConfigurationFactory$Factory.getConfiguration(ConfigurationFactory.java:536)
        at org.apache.logging.log4j.core.config.ConfigurationFactory$Factory.getConfiguration(ConfigurationFactory.java:457)
        at org.apache.logging.log4j.core.config.ConfigurationFactory.getConfiguration(ConfigurationFactory.java:321)
        at org.apache.logging.log4j.core.LoggerContext.reconfigure(LoggerContext.java:778)
        at org.apache.logging.log4j.core.LoggerContext.reconfigure(LoggerContext.java:808)
......
```

Ok, this becomes frustrating. I don't know how others achieved to build native-image from their projects, but if mine, witch have only couple of MB it is such a pain to build it, I'm very frustrated of this process. 

I just really want to see a project which is finally built as native-image with GraalVM, really. Is there any knew project on web? 🤔

**Comment by valimaties** [comment]
Finally I found out, after these years, that all those errors are generated because of using SLF4J library to log. I gave up using this library, I probably will find another way to log in app. There is no way to manage this library in native image, so... that's it!
That being said, I have managed to successfully build the native image. Now I'm getting another error, about sound playing, but this is another issue, which is not anymore related to this one here (much more to #9620 - I left some comments there), so in couple of days I will close this issue, as it is being solved.
FYI, @vjovanov 

BR,
Vali
